sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentStreamFactoryTest.java

```diff
@@ -0,0 +1,370 @@
+ */
+
+package com.cloudant.sync.datastore;
+
+import com.cloudant.sync.datastore.encryption.EncryptedAttachmentInputStream;
+import com.cloudant.sync.datastore.encryption.EncryptionTestConstants;
+import com.cloudant.sync.datastore.encryption.NullKeyProvider;
+import com.cloudant.sync.util.TestUtils;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Test the AttachmentStreamFactory for encrypted and unencrypted streams.
+ */
+public class AttachmentStreamFactoryTest extends BasicDatastoreTestBase {
+
+    // =======================
+    // Unencrypted, un-encoded
+    // =======================
+
+    @Test
+    /**
+     * Assert passing an AttachmentStreamFactory with no key writes an unencrypted file to disk.
+     */
+    public void testPreparedAttachmentWritesUnencryptedUnencodedStream()
+            throws AttachmentException, IOException {
+
+        AttachmentStreamFactory asf = new AttachmentStreamFactory(new NullKeyProvider());
+
+        File plainText = f("fixture/EncryptedAttachmentTest_plainText");
+
+        UnsavedFileAttachment usf = new UnsavedFileAttachment(plainText, "text/plain");
+        PreparedAttachment preparedAttachment = new PreparedAttachment(
+                usf, datastore_manager_dir, asf);
+
+        Assert.assertTrue("Writing to unencrypted, un-encoded stream didn't give correct output",
+                IOUtils.contentEquals(
+                        new FileInputStream(preparedAttachment.tempFile),
+                        new FileInputStream(plainText)));
+
+    }
+
+    @Test
+    /**
+     * Assert passing an AttachmentStreamFactory with no key reads an unencrypted file correctly.
+     */
+    public void testSavedAttachmentReadsUnencryptedUnencodedStream()
+            throws AttachmentException, IOException {
+
+        AttachmentStreamFactory asf = new AttachmentStreamFactory(new NullKeyProvider());
+
+        File plainText = f("fixture/EncryptedAttachmentTest_plainText");
+        SavedAttachment savedAttachment = new SavedAttachment("test", 0, 0, null, "text/plain",
+                plainText, Attachment.Encoding.Plain, asf);
+
+        Assert.assertTrue("Reading unencrypted, un-encoded stream didn't give correct output",
+                IOUtils.contentEquals(
+                        savedAttachment.getInputStream(),
+                        new FileInputStream(plainText)));
+
+    }
+
+    @Test
+    /**
+     * Assert passing an AttachmentStreamFactory with no key to both a PreparedAttachment (writer)
+     * and SavedAttachment (reader) round trips the data correctly.
+     */
+    public void testSavedAttachmentReadsPreparedAttachmentUnencryptedUnencodedStream()
```

Comment:
This test doesn't appear to be doing what it claims in the comment. It doesn't appear to round trip the data.